controller/src/beerocks/master/db/network_map.cpp

```diff
@@ -476,15 +539,21 @@ void network_map::send_bml_bss_tm_req_message_to_listeners(db &database,
         return;
     }
 
-    auto event = reinterpret_cast<BML_EVENT *>(response->buffer(0));
-    if (event == nullptr) {
+    auto buf = response->buffer(0);
+    if (!buf) {
+        LOG(ERROR) << "buffer is nullptr";
+        return;
+    }
+
+    auto event = reinterpret_cast<BML_EVENT *>(buf);
+    if (!event) {
         LOG(ERROR) << "event is nullptr";
         return;
     }
 
     event->type = BML_EVENT_TYPE_BSS_TM_REQ;
     auto size   = sizeof(BML_EVENT);
-    event->data = GET_MESSAGE_POINTER(BML_EVENT_BSS_TM_REQ, response->buffer(0), size);
+    event->data = GET_MESSAGE_POINTER(BML_EVENT_BSS_TM_REQ, buf, size);
```

Comment:
Meh. This stuff was ugly already, and it doesn't get better after the change... Why aren't bml events generated like the rest of the code? @morantr @vitalybu 

At least, if it is not generated, it would be better to add a helper template function that does all of the parsing and checking. And get rid of the GET_MESSAGE_POINTER macro.